text/0162-refactoring-support.md

```diff
@@ -0,0 +1,677 @@
+  If you only want to see what changes would be made, use the `--dry-run` flag.
+- If you pass any filters to the `deploy` command, the refactor will work on
+  those stacks plus any other stacks the refactor touches. For example, if you
+  choose to only deploy stack A, and a resource was moved from stack A to stack
+  B, the refactor will involve both stacks. But if there was a rename in, let's
+  say, stack C, it will not be refactored. The same set of filters is available
+  for the `refactor` command.
+- To perform refactoring, the CLI needs new permissions in the bootstrap stack.
+  Before using this feature, run `cdk bootstrap` for every target environment,
+  to add these new permissions.
+- A given CDK application can have multiple stacks, for different environments.
+  In that case, the CLI will group the stacks by environment and perform the
+  refactoring separately in each one. So, although you can move resources
+  between stacks, both stacks involved in the move must be in the same
+  environment. Trying to move resources across environments will result in an
+  error.
+
+### Explicit mapping
+
+Although the CLI will automatically refactor only what is unambiguous, you may
+still need to have more control over the refactoring process. Companies usually
+have stringent policies on how changes are made to the production environment,
+for example. It could be that your company requires that every change must be
+explicitly declared in some sort of code, including refactors (as opposed to
+computed on-the-fly).
+
+In a situation like this, you can import and export mapping files. Here is how
+it works: at development time, you made a change that the CLI detected as a
+refactor. Since you want that refactor to be propagated to other environments,
+you export the mapping file, with the command
+`cdk refactor --export-mapping=file.json`.
+
+There are at least two possible paths from here, depending on your constraints:
+
+1. You send the exported file to an operations team, who will review it. If
+   approved, they manually run the command
+   `cdk refactor --import-mapping=file.json` on every protected environment in
+   advance (i.e., before your changes get deployed to those environments). When
+   you import a mapping, the CLI won't try to detect refactors.
+2. The `--apply-mapping` option is also available for the `deploy` command. So
+   you can commit the mapping file to version control, and configure your
+   pipeline to use it on every deployment. This is a more convenient option, as
+   it requires less coordination between different roles. Every time a refactor
+   is applied, a record of it is stored in the environment. This is to prevent
+   the same refactor from being applied multiple times.
+
+In general, if the protected environment is not in the same state as the
+environment where the mapping was generated, the `refactor --apply-mapping`
+command will fail.
```

Comment:
How will it fail? what are the implications of failing? This seems like a scary touch point we need to clarify.